plugins/avro-format/core/src/main/java/org/pentaho/di/trans/steps/avro/output/PentahoAvroOutputFormat.java

```diff
@@ -242,4 +251,67 @@ public void setVariableSpace( VariableSpace variableSpace ) {
     this.variableSpace = variableSpace;
   }
 
+  private Object mapDefaultValuesToDataTypes( String defaultValue, AvroSpec.DataType type ) {
+    try {
+      switch ( type ) {
+        case BOOLEAN:
+          return Boolean.parseBoolean( defaultValue );
+        case DATE:
+          return getDateTypeForDefaultValue( defaultValue );
+        case FLOAT:
+          return Float.parseFloat( defaultValue );
+        case DOUBLE:
+          return Double.parseDouble( defaultValue );
+        case LONG:
+        case INTEGER:
+          return Long.parseLong( defaultValue );
+        case DECIMAL:
+          return getDecimalTypeForDefaultValue( defaultValue );
+        case STRING:
+          return String.valueOf( defaultValue );
+        case BYTES:
+          return defaultValue.getBytes();
+        case TIMESTAMP_MILLIS:
+          return getTimeStampTypeForDefaultValue( defaultValue );
+        default:
+          return defaultValue;
+      }
+    } catch ( NumberFormatException | ParseException exception ) {
+      throw new IllegalArgumentException( "Not a valid default value " + defaultValue + " for data type " + type + " - " + exception.getMessage());
+    }
+
+  }
+  private byte[] getDecimalTypeForDefaultValue( String defaultValue ) {
+    BigDecimal bigDecimalValue = new BigDecimal( defaultValue );
+    return bigDecimalValue.unscaledValue().toByteArray();
+  }
+
+  private int getDateTypeForDefaultValue( String defaultValue ) throws ParseException {
+    Date defaultDate;
+    DateFormat dateFormat = new SimpleDateFormat( ValueMetaBase.DEFAULT_DATE_PARSE_MASK );
+
+    try {
+      defaultDate = dateFormat.parse( defaultValue );
+      LocalDate localDate = defaultDate.toInstant()
+              .atZone( ZoneId.systemDefault() )
+              .toLocalDate();
+      return (int) localDate.toEpochDay();
+    } catch ( ParseException pe ) {
+      throw new ParseException( pe.getMessage(), pe.getErrorOffset() );
```

Comment:
Code change not necessary,

For `#getDateTypeForDefaultValue` this ParseException we are throwing, do we need to add a little bit more information, then just using the caught exception?

It seems we are adding some extra logic that can cause an exception like setting zone and local date.

Reply:
Addressed. Added more specific exception message.